### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

build failing at https://build.opensuse.org/package/live_build_log/openSUSE:Factory:Staging:G/yast2-packager/standard/x86_64

## Solution

Fix test to use proper string instead of Pathname.

## Testing

- *Tested manually*

### CODE REVIEW

File: src/lib/packager/product_patterns.rb [hunk 1]
```diff
@@ -99,8 +99,6 @@ def dependencies(product)
         end
       end
 
-      log.debug "Product #{product} depependencies: #{product_dependencies}"
```
Comment (mvidner) [inline src/lib/packager/product_patterns.rb:102]: You are smuggling in an undocumented change, and I have a better one to replace it :smile:

**Comment by mvidner** [comment]
https://github.com/yast/yast-ruby-bindings/blob/9a50d95be89f3dfe81cd109e6bf89a5fbedc48eb/src/ruby/yast/y2logger.rb#L48 shows that we're using a standard logger, and the [Ruby documentation shows](https://docs.ruby-lang.org/en/master/Logger.html#class-Logger-label-Entries) that passing a block will avoid evaluating it when debugging is disabled.

So please use `log.debug { "nasty string #{mem_bomb}" }` instead

BTW that Ruby docs is so skillful in hiding the useful information, it sucks

**Comment by jreidinger** [comment]
@mvidner sadly block form does not work, as we do not limit access to debug in logger level but on ruby-bindings level

try yourself in irb:
```
1 require "yast"
  2 
  3 include Yast::Logger
  4 
  5 log.debug { puts "eval"; "log content" }
```

**Comment by mvidner** [comment]
Ah, good point. Then what about

1. make an Issue in ruby-bindings "log.debug { expensive } always evaluates the block"
2. leave the calls here, commented out, referencing that issue

**Comment by jreidinger** [comment]
issue created https://github.com/yast/yast-ruby-bindings/issues/290

**Review by mvidner — COMMENTED** [review]
Fixes so far look good, but what about
- the other failure "unexpected message :define_method with ("switch_to_richtext=")", do you need help?
- version + changelog?

File: test/addon_selector_test.rb [hunk 2]
```diff
@@ -66,7 +66,8 @@
 
       it "does not display any popup" do
         # stub empty Yast::Popup so any method call would raise an exception
-        stub_const("Yast::Popup", double)
```
Comment (mvidner) [inline test/addon_selector_test.rb:69]: Does that mean that any such `stub_const("Yast::Foo", double)` will fail or has bug? We have more

https://github.com/search?q=org%3Ayast+stub_const&type=code

Comment (mvidner) [reply test/addon_selector_test.rb:69]: yast/yast-update/test/inst_update_partition_auto_test.rb
15:      stub_const("Yast::FileSystems", double)
yast/yast-installation/test/snapshots_finish_test.rb
10:    stub_const("Yast::StorageSnapper", double)

**Review by mvidner — APPROVED** [review]
Thanks!

I think there is still a chance that another fix will be needed in ruby-bindings because of the following, but I will not let that block this PR.

> Does that mean that any such `stub_const("Yast::Foo", double)` will fail or has bug? We have more: https://github.com/search?q=org%3Ayast+stub_const&type=code

**Comment by mvidner** [comment]
For the record, the `stub_const` fix adjusts for the change done in https://github.com/yast/yast-ruby-bindings/pull/289

**Pull request merged**